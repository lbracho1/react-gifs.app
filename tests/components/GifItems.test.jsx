import { render, screen } from "@testing-library/react"
import { GifItems } from "../../src/components/GifItems"


describe('Pruebas en GifItems', () => {

    const title = 'Toyota Hilux';
    const url = 'https://www.tucarro.com.ve/';

    test('debe de hacer match con el snapshot', () => {

        const { container } = render( <GifItems title={ title } url={ url } /> );
        expect( container ).toMatchSnapshot();
        
    });

    test('debe de mostar las pruebas con el URL y el ALT indicado', () => {
        
        render( <GifItems title={ title } url={ url } /> );
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    })


    test('debe de mostar el titulo en el componente', () => { 
       
        render( <GifItems title={ title } url={ url } /> );
        expect( screen.getByText( title ) ).toBeTruthy();

     })


    })