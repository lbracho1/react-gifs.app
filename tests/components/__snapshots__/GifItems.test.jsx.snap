// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItems debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Toyota Hilux"
      src="https://www.tucarro.com.ve/"
    />
    <p>
      Toyota Hilux
    </p>
  </div>
</div>
`;
